### PR TITLE
Adds queryOptions to type definition for codegen output

### PR DESCRIPTION
**Pull request by noahlitvin**

### CODE REVIEW

**Comment by fritzschoff** [comment]
You also getting an error while building the queries package? 
`
The inferred type of "X" cannot be named without a reference to "Y". This is likely not portable. A type annotation is necessary.
`

**Comment by noahlitvin** [comment]
Yeah something weird is going on. I think your issue on #129 is unrelated, but I'm not sure.

File: packages/queries/src/index.ts [hunk 1]
```diff
@@ -32,12 +32,22 @@ type OmitFirstArg<F> = F extends (x: any, ...args: infer P) => infer R ? (...arg
 
 type SubgraphQueryFunction<F> = F extends (
 	u: string,
-	o: exchanges.MultiQueryOptions<infer A, infer B>,
-	args: infer C
+	o: subgraph.MultiQueryOptions<infer A, infer B>,
+	args: infer C,
+	queryOptions: UseQueryOptions<Pick<infer Result, infer K>[]>
```
Comment (dbeal-eth) [inline packages/queries/src/index.ts:37]: this is supposed to be `UseQueryOptions<infer R>`

sorry

File: packages/queries/src/index.ts [hunk 2]
```diff
@@ -32,12 +32,22 @@ type OmitFirstArg<F> = F extends (x: any, ...args: infer P) => infer R ? (...arg
...
 ) => infer R
-	? (options: exchanges.MultiQueryOptions<A, B>, args: Partial<C>) => R
-	: F extends (u: string, o: exchanges.SingleQueryOptions, args: infer A) => infer P
-	? (options: exchanges.SingleQueryOptions, args: Partial<A>) => P
+	? (
+			options: subgraph.MultiQueryOptions<A, B>,
+			args: Partial<C>,
+			queryOptions?: UseQueryOptions<Pick<Result, K>[]>
```
Comment (dbeal-eth) [inline packages/queries/src/index.ts:42]: same here (and all following)

File: packages/queries/src/index.ts [hunk 3]
```diff
@@ -32,12 +32,22 @@ type OmitFirstArg<F> = F extends (x: any, ...args: infer P) => infer R ? (...arg
...
+	  ) => R
+	: F extends (
+			u: string,
+			o: subgraph.SingleQueryOptions,
+			args: infer A,
+			queryOptions?: UseQueryOptions
+	  ) => infer P
+	? (options: subgraph.SingleQueryOptions, args: Partial<A>) => P
```
Comment (dbeal-eth) [inline packages/queries/src/index.ts:50]: also its missing here it looks like

**Review by dbeal-eth — APPROVED** [review]
thanks for fixing

**Pull request merged**